### PR TITLE
Remove 'SGTK_CONFIG_LOCK_VERSION' warning on tk-framework-desktopstartup.

### PR DESCRIPTION
**Pull request by NorberMV**
This PR removes the 'SGTK_CONFIG_LOCK_VERSION' environment variable check on tk-framework-desktopstartup, and updates tk-core to v0.20.16.

### CODE REVIEW

**Review by 000paradox000 — APPROVED** [review]
LGTM

File: python/tk-core/python/tank/bootstrap/resolver.py [hunk 1]
```diff
@@ -206,31 +206,30 @@ def resolve_not_found_sg_configuration(self, config_descriptor, sg_connection):
         # Validate if descriptor version token is omitted
         resolve_latest = is_descriptor_version_missing(config_descriptor)
         if (
-                constants.SGTK_CONFIG_LOCK_VERSION in os.environ
-                and sys.version_info[0] != 3
+                sys.version_info[0] < 3
```
Comment (julien-lang) [inline python/tk-core/python/tank/bootstrap/resolver.py:209]: `and` missing here

Comment (NorberMV) [reply python/tk-core/python/tank/bootstrap/resolver.py:209]: It is for the "unified view" of Github, here I show a split view:
Screen Shot 2023-01-12 at 2 24 54 PM

Comment (julien-lang) [reply python/tk-core/python/tank/bootstrap/resolver.py:209]: Yep, I missed that. Sorry, you're right.

**Review by 000paradox000 — APPROVED** [review]
LGTM

**Pull request merged**